src/lib/relicScorer.ts

```diff
@@ -425,65 +386,147 @@ export class RelicScorer {
+      , fakesubs[0].stat, fakesubs[0].value, fakesubs[1].stat, fakesubs[1].value
+      , fakesubs[2].stat, fakesubs[2].value, fakesubs[3].stat, fakesubs[3].value,
+    )), id).longscore) + mainScoreDeduction
+
+    // Calculate the average score
+    for (let i = 0; i < relic.substats.length; i++) { // Carry over the already existing substats and add the average number of mid rolls to the value
+      fakesubs[i].stat = relic.substats[i.toString()].stat
+      fakesubs[i].value = relic.substats[i.toString()].value + remainingRolls / 4 * SubStatValues[fakesubs[i].stat][relic.grade].mid
+    }
+    // We want to use the score() function to score relics for maximum accuracy (and easier maintainability potentially)
+    // How do we score a relic with unknown substats?
+    // One option would be to score all possible relics and take the average score (ew, slow, expensive)
+    // Instead, we calculate the average score of the possible new line (using a mid roll)
+    // We then choose a substat to camouflage as so that the score() function is able to handle the relic
+    // We divide the average score of 1 mid roll we calculated earlier by the weight and normalization of the chosen camouflage stat
+    // We then multiply this by the number of rolls our filler stat will have
+    // This way, when the score() function evaluates our relic, all the additional lines will score their expected average score
+    let averageScore: number = 0
+    const scaling = {
+      [Constants.Stats.HP_P]: 64.8 / 43.2,
+      [Constants.Stats.ATK_P]: 64.8 / 43.2,
+      [Constants.Stats.DEF_P]: 64.8 / 54,
+      [Constants.Stats.HP]: 1 / (DB.getMetadata().characters[id].promotions[80][Constants.Stats.HP] * 2 * 0.01) * (64.8 / 43.2),
+      [Constants.Stats.ATK]: 1 / (DB.getMetadata().characters[id].promotions[80][Constants.Stats.ATK] * 2 * 0.01) * (64.8 / 43.2),
+      [Constants.Stats.DEF]: 1 / (DB.getMetadata().characters[id].promotions[80][Constants.Stats.DEF] * 2 * 0.01) * (64.8 / 54),
+      [Constants.Stats.CR]: 64.8 / 32.4,
+      [Constants.Stats.CD]: 64.8 / 64.8,
+      [Constants.Stats.OHB]: 64.8 / 34.5606,
+      [Constants.Stats.EHR]: 64.8 / 43.2,
+      [Constants.Stats.RES]: 64.8 / 43.2,
+      [Constants.Stats.SPD]: 64.8 / 25.032,
+      [Constants.Stats.BE]: 64.8 / 64.8,
+    }
+    for (let i = 0; i < remainingSubStats.length; i++) {
+      averageScore += remainingSubStats[i][1] * scaling[remainingSubStats[i][0]] * SubStatValues[remainingSubStats[i][0]][relic.grade].mid
+    }
+    averageScore = averageScore / remainingSubStats.length
+    // average score of 1 midroll = avg(weight * normalization * midroll)
+    // avg score = avg(weight * norm * midroll) * rolls
+    // acg score = value * weight * scaling
+    // value = avg(weight * norm * midroll) * rolls / (weight * scaling)
+    // value = average score of 1 midroll * rolls / (weight * scaling)
```

Comment:
Some cleanup for dead code

Reply:
This was just writing out the logic for why I assign the value of the substat below in the way I do, I'll add a small note above this to make that clearer